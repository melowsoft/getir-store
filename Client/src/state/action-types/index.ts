export enum ActionType {
  FETCH_PRODUCTS = 'fetch_products',
  FETCH_PRODUCTS_SUCCESS = 'fetch_products_success',
  FETCH_PRODUCTS_ERROR = 'fetch_products_error',
  ADD_TO_BASKET = 'add_to_basket',
  REMOVE_FROM_BASKET = 'remove_from_basket',
  TOGGLE_BASKET = 'toggle_basket',
  TOGGLE_SIDE_FILTER = 'toggle_side_filter',
}