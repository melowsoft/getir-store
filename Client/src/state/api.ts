

const {
  REACT_APP_API_DEV_BASE_URL,
  REACT_APP_API_PROD_BASE_URL,
  NODE_ENV
} = process.env

export const BASE_URL: any = NODE_ENV === "production"
? REACT_APP_API_PROD_BASE_URL
: REACT_APP_API_DEV_BASE_URL