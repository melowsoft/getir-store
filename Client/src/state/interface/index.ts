export interface Product {
  tags: string[];
  price: number;
  name: string;
  description: string;
  slug: string;
  added: number;
  manufacturer: string;
  itemType: string;
}

export interface BasketPick {
  name: string;
  price: number;
  quantity: number;
}

export interface CartItem {
  name: string;
  price: number;
}