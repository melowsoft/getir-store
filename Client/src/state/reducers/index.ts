import { combineReducers } from 'redux'
import  productsReducer from './productsReducers'
import basketReducer from "./basketReducers"
import showBasketReducer from "./showBasketReducer"
import showSideFilterReducer from "./showSideFilterReducer"

const reducers = combineReducers({
  products: productsReducer,
  basket: basketReducer,
  showBasket: showBasketReducer,
  showSideFilter: showSideFilterReducer
});

export default reducers;

export type RootState = ReturnType<typeof reducers>