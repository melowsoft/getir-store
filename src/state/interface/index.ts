export interface Product {
  tags: string[];
  price: 10.99;
  name: string;
  description: string;
  slug: string;
  added: number;
  manufacturer: string;
  itemType: string;
}

export interface BasketPick {
  name: string;
  price: number;
  quantity: number;
}

export interface CartItem {
  name: string;
  price: number;
}