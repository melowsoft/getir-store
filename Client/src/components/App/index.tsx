import {Provider} from 'react-redux';
import { store } from '../../state'
import {Container} from "./AppStyles"
import { BrandIcon } from '../BrandIcon';
import { Navbar } from '../Navbar';
import { ContentWrapper } from '../ContentWrapper';
import { ContentArea } from '../ContentArea';


const App = () => {
  return (
  <Provider store={store}>
    <Container>
      <Navbar icon={<BrandIcon />}/>
      <ContentWrapper>
        <ContentArea />
      </ContentWrapper>
    </Container>
  </Provider>
  );
  
}

export default App;
