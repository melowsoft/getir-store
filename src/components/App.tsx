import {Provider} from 'react-redux';
import { store } from '../state'


import {Container} from "./AppStyles"
import { BrandIcon } from './BrandIcon';
import { Navbar } from './Navbar';
import { ContentWrapper } from './ContentWrapper';


const App = () => {
  return <Provider store={store}>
    <Container>
      <Navbar icon={<BrandIcon />}/>
      <ContentWrapper >
        <h1>Hello from Wrapper</h1>
      </ContentWrapper>
    </Container>
  </Provider>;
  
}

export default App;
