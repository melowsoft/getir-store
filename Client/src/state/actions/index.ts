import { ActionType } from "../action-types"
import {CartItem, Product} from "../interface"

interface FetchProductsAction {
  type: ActionType.FETCH_PRODUCTS;
}
interface ToggleBasketAction {
  type: ActionType.TOGGLE_BASKET;
}
interface ToggleSideFilterAction {
  type: ActionType.TOGGLE_SIDE_FILTER;
  payload: string;
}
interface FetchProductsSuccessAction {
  type: ActionType.FETCH_PRODUCTS_SUCCESS;
  payload: {products: Product[], pagination: {totalPages: number}, tags: string[], brands: string[]};
}
interface FetchProductsErrorAction {
  type: ActionType.FETCH_PRODUCTS_ERROR;
  payload: string;
}

interface AddProductToBasket {
  type: ActionType.ADD_TO_BASKET;
  payload: CartItem;
}
interface RemoveProductFromBasket {
  type: ActionType.REMOVE_FROM_BASKET;
  payload: CartItem;
}

export type Action =   
| FetchProductsAction 
| FetchProductsSuccessAction 
| FetchProductsErrorAction
| AddProductToBasket
| RemoveProductFromBasket
| ToggleBasketAction
| ToggleSideFilterAction
