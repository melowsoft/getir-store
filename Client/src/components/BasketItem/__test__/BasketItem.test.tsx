import React from "react"
import  ReactDOM  from "react-dom"
import { BasketItem } from "..";
import { BasketPick} from "../../../state/interface"
import { cleanup } from '@testing-library/react';
import renderer from "react-test-renderer"
import {Provider} from 'react-redux';
import App from "../../App"
import { store } from '../../../state'


const product:BasketPick = {
  price: 10,
  name: "test name",
  quantity: 0
}

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Provider store={store}><App><BasketItem product={product}></BasketItem></App></Provider>, div);
  ReactDOM.unmountComponentAtNode(div)
})


it("matches snapshot", () => {
  const tree = renderer.create(<Provider store={store}><App><BasketItem product={product}></BasketItem></App></Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})