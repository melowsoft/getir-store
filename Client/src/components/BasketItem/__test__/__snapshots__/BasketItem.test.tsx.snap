// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-iCoGMd dYWLQi"
>
  <div
    className="sc-jrsJWt kalrgg"
  >
    <div
      className="sc-crzoAE hfmyoP close-btn"
      onClick={[Function]}
    >
      <img
        alt="filter icon"
        className="sc-dIsUp eoWJNn"
        src="filter-icon.svg"
      />
    </div>
    <div
      className="sc-fujyAs eDHmge"
      data-testid="brand-logo"
    >
      <img
        alt="brand-logo"
        className="sc-pNWdM gMSCoc"
        src="Logo.svg"
      />
    </div>
    <div
      className="sc-kEqXSa bDwqqy"
      onClick={[Function]}
    >
      <svg>
        basket.svg
      </svg>
      <p
        className="sc-iqAclL giIpgf"
      >
        â‚º 
        0
      </p>
    </div>
  </div>
  <div
    className="sc-bqGGPW gFFraJ"
  >
    <div
      className="sc-Arkif fGJwhU"
    >
      <div
        className="sc-khIgEk CvPyH menu"
      >
        <div
          className="sc-hTRkXV kyjcmo"
          onClick={[Function]}
        >
          x
        </div>
        <div
          className="sc-cxNHIi"
        >
          <p
            className="sc-iemWCZ jtakIy"
          >
            Sorting
          </p>
          <div
            className="sc-dIvrsQ iwFhlO"
          >
            <div
              className="sc-hHEiqL dHgoxS"
            >
              <div
                className="sc-giAqHp kRJUrK"
              >
                <div
                  className="sc-lmgQwP fpirhD"
                  onClick={[Function]}
                >
                  <svg>
                    selected-circle.svg
                  </svg>
                  <p
                    className="sc-iJCRrE bDJUT"
                  >
                    Price low to high
                  </p>
                </div>
                <div
                  className="sc-lmgQwP fpirhD"
                  onClick={[Function]}
                >
                  <svg>
                    unselected-circle.svg
                  </svg>
                  <p
                    className="sc-iJCRrE bDJUT"
                  >
                    Price high to low
                  </p>
                </div>
                <div
                  className="sc-lmgQwP fpirhD"
                  onClick={[Function]}
                >
                  <svg>
                    unselected-circle.svg
                  </svg>
                  <p
                    className="sc-iJCRrE bDJUT"
                  >
                    New to old
                  </p>
                </div>
                <div
                  className="sc-lmgQwP fpirhD"
                  onClick={[Function]}
                >
                  <svg>
                    unselected-circle.svg
                  </svg>
                  <p
                    className="sc-iJCRrE bDJUT"
                  >
                    Old to new
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-dlMDgC hjEWxQ"
        >
          <p
            className="sc-iemWCZ jtakIy"
          >
            Brands
          </p>
          <div
            className="sc-dIvrsQ gBctAa"
            height={230}
          >
            <div
              className="sc-hHEiqL dHgoxS"
            >
              <input
                className="sc-kfYoZR iFXTJT"
                onChange={[Function]}
                placeholder="Search Brands"
                type="text"
                value=""
              />
              <div
                className="sc-iwajpm jeqwbY"
              >
                <div
                  className="sc-fKgJPI dhubWG"
                  onClick={[Function]}
                >
                  <svg>
                    selected-box.svg
                  </svg>
                  <p
                    className="sc-bCwfaz cGFLA-d"
                  >
                    ALL (
                    0
                    )
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-iklJeh eZZNhe"
        >
          <p
            className="sc-iemWCZ jtakIy"
          >
            Tags
          </p>
          <div
            className="sc-dIvrsQ gBctAa"
            height={230}
          >
            <div
              className="sc-hHEiqL dHgoxS"
            >
              <input
                className="sc-kfYoZR iFXTJT"
                onChange={[Function]}
                placeholder="Search Tags"
                type="text"
                value=""
              />
              <div
                className="sc-kLojOw fQeGFO"
              >
                <div
                  className="sc-ezzafa cjpNJE"
                  onClick={[Function]}
                >
                  <svg>
                    selected-box.svg
                  </svg>
                  <p
                    className="sc-bYwzuL brwxbB"
                  >
                    ALL (
                    0
                    )
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-jJMGnK hkzeAK"
      >
        <div
          className="sc-cxNHIi"
        >
          <p
            className="sc-iemWCZ jtakIy"
          >
            Sorting
          </p>
          <div
            className="sc-dIvrsQ iwFhlO"
          >
            <div
              className="sc-hHEiqL dHgoxS"
            >
              <div
                className="sc-giAqHp kRJUrK"
              >
                <div
                  className="sc-lmgQwP fpirhD"
                  onClick={[Function]}
                >
                  <svg>
                    selected-circle.svg
                  </svg>
                  <p
                    className="sc-iJCRrE bDJUT"
                  >
                    Price low to high
                  </p>
                </div>
                <div
                  className="sc-lmgQwP fpirhD"
                  onClick={[Function]}
                >
                  <svg>
                    unselected-circle.svg
                  </svg>
                  <p
                    className="sc-iJCRrE bDJUT"
                  >
                    Price high to low
                  </p>
                </div>
                <div
                  className="sc-lmgQwP fpirhD"
                  onClick={[Function]}
                >
                  <svg>
                    unselected-circle.svg
                  </svg>
                  <p
                    className="sc-iJCRrE bDJUT"
                  >
                    New to old
                  </p>
                </div>
                <div
                  className="sc-lmgQwP fpirhD"
                  onClick={[Function]}
                >
                  <svg>
                    unselected-circle.svg
                  </svg>
                  <p
                    className="sc-iJCRrE bDJUT"
                  >
                    Old to new
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-dlMDgC hjEWxQ"
        >
          <p
            className="sc-iemWCZ jtakIy"
          >
            Brands
          </p>
          <div
            className="sc-dIvrsQ gBctAa"
            height={230}
          >
            <div
              className="sc-hHEiqL dHgoxS"
            >
              <input
                className="sc-kfYoZR iFXTJT"
                onChange={[Function]}
                placeholder="Search Brands"
                type="text"
                value=""
              />
              <div
                className="sc-iwajpm jeqwbY"
              >
                <div
                  className="sc-fKgJPI dhubWG"
                  onClick={[Function]}
                >
                  <svg>
                    selected-box.svg
                  </svg>
                  <p
                    className="sc-bCwfaz cGFLA-d"
                  >
                    ALL (
                    0
                    )
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-iklJeh eZZNhe"
        >
          <p
            className="sc-iemWCZ jtakIy"
          >
            Tags
          </p>
          <div
            className="sc-dIvrsQ gBctAa"
            height={230}
          >
            <div
              className="sc-hHEiqL dHgoxS"
            >
              <input
                className="sc-kfYoZR iFXTJT"
                onChange={[Function]}
                placeholder="Search Tags"
                type="text"
                value=""
              />
              <div
                className="sc-kLojOw fQeGFO"
              >
                <div
                  className="sc-ezzafa cjpNJE"
                  onClick={[Function]}
                >
                  <svg>
                    selected-box.svg
                  </svg>
                  <p
                    className="sc-bYwzuL brwxbB"
                  >
                    ALL (
                    0
                    )
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-dPaNzc iHIBaR"
      >
        <h4
          className="sc-hiKfDv fYrDsz"
        >
          Products
        </h4>
        <div
          className="sc-bBjRSN fnoqUv"
        >
          <div
            className="sc-cOifOu fMMFpb"
            onClick={[Function]}
          >
            Mug
          </div>
          <div
            className="sc-cOifOu bTLRiT"
            onClick={[Function]}
          >
            Shirt
          </div>
        </div>
        <div
          className="sc-jNnpgg giAWqf"
        />
        <div
          className="sc-gXfVKN kDxCti"
        >
          <ul
            className="pagination"
          >
            <li
              className="previous disabled"
            >
              <a
                aria-disabled="true"
                aria-label="Previous page"
                className="prev-link-class"
                onClick={[Function]}
                onKeyPress={[Function]}
                rel="prev"
                role="button"
                tabIndex="0"
              >
                <div
                  className="sc-ciSkZP gfaCBS"
                >
                  <svg>
                    prev.svg
                  </svg>
                   
                  <div>
                    Prev
                  </div>
                </div>
              </a>
            </li>
            <li
              className="next"
            >
              <a
                aria-disabled="false"
                aria-label="Next page"
                className="next-link-class"
                onClick={[Function]}
                onKeyPress={[Function]}
                rel="next"
                role="button"
                tabIndex="0"
              >
                <div
                  className="sc-cBoqAE bHmcrx"
                >
                  <div>
                    Next
                  </div>
                   
                  <svg>
                    next.svg
                  </svg>
                </div>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="sc-ksluID kypamZ"
        data-testid="basket-box"
      />
    </div>
  </div>
</div>
`;
